src/odemis/driver/hamamatsurx.py

```diff
@@ -641,6 +646,36 @@ def __init__(self, name, role, parent, daemon=None, **kwargs):
 
         # read-only VAs TODO: Trig. Mode, Trig. level, Trig. slope?
 
+    def _updateSettings(self) -> None:
+        """
+        Read all the current streak unit settings from the RemoteEx and reflect them on the VAs
+        """
+        logging.debug("Updating streak unit settings")
+        try:
+            timeRange = self.GetTimeRange()
+            if timeRange != self.timeRange.value:
+                self.timeRange._value = timeRange
+                self.timeRange.notify(timeRange)
```

Comment:
For several VAs, you'd need to update the metadata (see line 617 - 619). Probably the simplest would be to separate the setter from the metadata update by placing the metadata update in a subscriber.